2005-02-08  Jakub Jelinek  <jakub@redhat.com>

	* src/alpha/osf.S: Add .note.GNU-stack on Linux.
	* src/s390/sysv.S: Likewise.
	* src/powerpc/linux64.S: Likewise.
	* src/powerpc/linux64_closure.S: Likewise.
	* src/powerpc/ppc_closure.S: Likewise.
	* src/powerpc/sysv.S: Likewise.
	* src/x86/unix64.S: Likewise.
	* src/x86/sysv.S: Likewise.
	* src/sparc/v8.S: Likewise.
	* src/sparc/v9.S: Likewise.
	* src/m68k/sysv.S: Likewise.
	* src/ia64/unix.S: Likewise.
	* src/arm/sysv.S: Likewise.

	* ia64_save_regs_in_stack.s: Moved to...
	* ia64_save_regs_in_stack.S: ... this.  Add .note.GNU-stack
	on Linux.

---
 boehm-gc/ia64_save_regs_in_stack.S   |   15 +++++++++++++++
 boehm-gc/ia64_save_regs_in_stack.s   |   12 ------------
 libffi/src/ia64/unix.S               |    4 ++++
 libffi/src/powerpc/linux64.S         |    4 ++++
 libffi/src/powerpc/linux64_closure.S |    4 ++++
 libffi/src/powerpc/ppc_closure.S     |    4 ++++
 libffi/src/powerpc/sysv.S            |    4 ++++
 libffi/src/sparc/v9.S                |    4 ++++
 8 files changed, 39 insertions(+), 12 deletions(-)

Index: gcc-4.4.0/boehm-gc/ia64_save_regs_in_stack.S
===================================================================
--- /dev/null
+++ gcc-4.4.0/boehm-gc/ia64_save_regs_in_stack.S
@@ -0,0 +1,15 @@
+        .text
+        .align 16
+        .global GC_save_regs_in_stack
+        .proc GC_save_regs_in_stack
+GC_save_regs_in_stack:
+        .bodyfoo.mpg
+        flushrs
+        ;;
+        mov r8=ar.bsp
+        br.ret.sptk.few rp
+        .endp GC_save_regs_in_stack
+
+#ifdef __linux__
+	.section .note.GNU-stack,"",@progbits
+#endif
Index: gcc-4.4.0/boehm-gc/ia64_save_regs_in_stack.s
===================================================================
--- gcc-4.4.0.orig/boehm-gc/ia64_save_regs_in_stack.s
+++ /dev/null
@@ -1,12 +0,0 @@
-        .text
-        .align 16
-        .global GC_save_regs_in_stack
-        .proc GC_save_regs_in_stack
-GC_save_regs_in_stack:
-        .body
-        flushrs
-        ;;
-        mov r8=ar.bsp
-        br.ret.sptk.few rp
-        .endp GC_save_regs_in_stack
-
Index: gcc-4.4.0/libffi/src/ia64/unix.S
===================================================================
--- gcc-4.4.0.orig/libffi/src/ia64/unix.S
+++ gcc-4.4.0/libffi/src/ia64/unix.S
@@ -553,3 +553,7 @@ ffi_closure_unix:
 	data8	@pcrel(.Lld_hfa_float)		// FFI_IA64_TYPE_HFA_FLOAT
 	data8	@pcrel(.Lld_hfa_double)		// FFI_IA64_TYPE_HFA_DOUBLE
 	data8	@pcrel(.Lld_hfa_ldouble)	// FFI_IA64_TYPE_HFA_LDOUBLE
+
+#if defined __ELF__ && defined __linux__
+	.section	.note.GNU-stack,"",@progbits
+#endif
Index: gcc-4.4.0/libffi/src/powerpc/linux64.S
===================================================================
--- gcc-4.4.0.orig/libffi/src/powerpc/linux64.S
+++ gcc-4.4.0/libffi/src/powerpc/linux64.S
@@ -179,3 +179,7 @@ ffi_call_LINUX64:
 	.align 3
 .LEFDE1:
 #endif
+
+#if defined __ELF__ && defined __linux__
+	.section	.note.GNU-stack,"",@progbits
+#endif
Index: gcc-4.4.0/libffi/src/powerpc/linux64_closure.S
===================================================================
--- gcc-4.4.0.orig/libffi/src/powerpc/linux64_closure.S
+++ gcc-4.4.0/libffi/src/powerpc/linux64_closure.S
@@ -204,3 +204,7 @@ ffi_closure_LINUX64:
 	.align 3
 .LEFDE1:
 #endif
+
+#if defined __ELF__ && defined __linux__
+	.section	.note.GNU-stack,"",@progbits
+#endif
Index: gcc-4.4.0/libffi/src/powerpc/ppc_closure.S
===================================================================
--- gcc-4.4.0.orig/libffi/src/powerpc/ppc_closure.S
+++ gcc-4.4.0/libffi/src/powerpc/ppc_closure.S
@@ -299,3 +299,7 @@ END(ffi_closure_SYSV)
 #if defined __ELF__ && defined __linux__
 	.section	.note.GNU-stack,"",@progbits
 #endif
+
+#if defined __ELF__ && defined __linux__
+	.section	.note.GNU-stack,"",@progbits
+#endif
Index: gcc-4.4.0/libffi/src/powerpc/sysv.S
===================================================================
--- gcc-4.4.0.orig/libffi/src/powerpc/sysv.S
+++ gcc-4.4.0/libffi/src/powerpc/sysv.S
@@ -227,3 +227,7 @@ END(ffi_call_SYSV)
 #if defined __ELF__ && defined __linux__
 	.section	.note.GNU-stack,"",@progbits
 #endif
+
+#if defined __ELF__ && defined __linux__
+	.section	.note.GNU-stack,"",@progbits
+#endif
Index: gcc-4.4.0/libffi/src/sparc/v9.S
===================================================================
--- gcc-4.4.0.orig/libffi/src/sparc/v9.S
+++ gcc-4.4.0/libffi/src/sparc/v9.S
@@ -304,3 +304,7 @@ longdouble1:
 #ifdef __linux__
 	.section	.note.GNU-stack,"",@progbits
 #endif
+
+#ifdef __linux__
+	.section	.note.GNU-stack,"",@progbits
+#endif
